addons/hr_timesheet/models/res_company.py

```diff
@@ -54,25 +49,51 @@ def create(self, values):
         return company
 
     def _create_internal_project_task(self):
-        results = []
-        type_ids = [(4, self.env.ref('hr_timesheet.internal_project_default_stage').id)]
-        for company in self:
-            company = company.with_company(company)
-            results += [{
+        project = self.env['project.project'].search([('is_internal_project', '=', True), ('active', '=', True), ('analytic_account_id', '!=', False),
+                                                      ('analytic_account_id.company_id', '=', False)], limit=1)
+
+        if not project:
+            default_plan = self.env['account.analytic.plan'].sudo().search([('company_id', '=', False)], limit=1)
+            if not default_plan:
+                default_plan = self.env['account.analytic.plan'].create({
+                    'name': 'Default',
+                    'company_id': False,
+                })
+            val = {
                 'name': _('Internal'),
                 'allow_timesheets': True,
-                'company_id': company.id,
-                'type_ids': type_ids,
+                'company_id': self.ids[0],
+                'type_ids': [(4, self.env.ref('hr_timesheet.internal_project_default_stage').id)],
                 'task_ids': [(0, 0, {
                     'name': name,
-                    'company_id': company.id,
+                    'company_id': self.ids[0],
                 }) for name in [_('Training'), _('Meeting')]]
-            }]
-        project_ids = self.env['project.project'].create(results)
-        projects_by_company = {project.company_id.id: project for project in project_ids}
+            }
+            project = self.env['project.project'].create(val)
+            project.analytic_account_id.plan_id = default_plan
+            project.analytic_account_id.company_id = False
+
+        for company in self:
+            company.internal_project_id = project.id
+        return project
+
+    def _inverse_is_internal_project(self):
```

Comment:
If we implement this in the `write` rather than using an inverse, finding the previous internal projects will be a lot easier